### ISSUE TITLE
Usage of `n()` and `$n()` with `undefined` should not throw errors

### ISSUE THREAD

**Issue by MickL** [intlify/vue-i18n-next#1746]
### Clear and concise description of the problem

Sometimes values are undefined within an application, unfortunately this will throw a 500 error (at least in Nuxt) so the whole page is unavailable just because one variable is undefined:

```
<script setup lang="ts">
const props = defineProps<{price: number}>();
</script>

<template>
{{ $n(props.price, 'currency') }}
</template>
```

Another example are inputs. These can get undefined while the user is editing the input value which will also throw errors:

> Incaught (in promise) SyntaxError: Invalid arguments (at message-compiler.mjs:77:19)

### Suggested solution

It would be great if the function `n()` and `$n()` would instead just return an empty string `''` without the whole page crashing or throwing any errors.

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/intlify/vue-i18n-next/blob/master/.github/CONTRIBUTING.md)
- [X] Read the [Documentation](https://vue-i18n.intlify.dev/)
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by kazupon** [comment]
Thank you for your feedback!

The first argument of `n` and `$n` must be `number`.
Don't pass `undefined`.
It is the responsibility of the application to verify the value passed.

However, vue-i18n API documentation does not mention that an error is raised if an unexpected value is specified.
https://vue-i18n.intlify.dev/api/composition.html#composernumberformatting

I think it would help to mention it in the documentation.

**Comment by MickL** [comment]
Hey, thanks for your answer! It would be great if it doesnt throw an error because values can always be undefined especially when working with inputs. And because this function just returns a string it would be nice if it returns an empty string in case of undefined input imo.

**Comment by MickL** [comment]
Can we please allow `n()` and `$n()` allow to have undefined values? Consider the second example from above:

- The first example above will break the full page
- The second example above will throw errors because price is getting and empty string when the input is empty

**Comment by KaelWD** [comment]
Also #1155

These strict runtime checks are incredibly inconvenient, I don't understand why you need to throw an error (often breaking the entire page) when just passing through null values and casting numbers to strings automatically would at worst display some mildly confusing strings to users and at best cause no problems because we have null checks *after* $t().

**Comment by milesstt** [comment]
I agree. When the key is a dynamic value, it can be expected behavior for it to be null and it shouldn't break the entire page. It's annoying to have to check for null in each case. Maybe log a warning instead of throwing an error.
